core/trino-main/src/main/java/io/trino/dispatcher/FailedDispatchQuery.java

```diff
@@ -262,6 +262,8 @@ private static QueryStats immediateFailureQueryStats()
                 new Duration(0, MILLISECONDS),
                 new Duration(0, MILLISECONDS),
                 new Duration(0, MILLISECONDS),
+                new Duration(0, MILLISECONDS),
```

Comment:
How `runningTime` is different than `io.trino.execution.QueryStateTimer#executionTime`?
Are the metrics already in `io.trino.execution.QueryStateTimer`?

Reply:
https://github.com/trinodb/trino/blob/51d91df5db92b4d51346f6b13b649b7568a2c972/core/trino-main/src/main/java/io/trino/event/QueryMonitor.java#L586

Reply:
Code in `io.trino.event.QueryMonitor#logQueryTimeline(io.trino.execution.QueryInfo)` should use metrics computed here.

Why `max` is used there?

Reply:
Yep, there is no need compute metrics dulplicated both in io.trino.event.QueryMonitor and  io.trino.execution.QueryStateMachine. Is that better remoe compute  logic from QueryMonitor